### PR TITLE
Add linter rule for RPC-Get-V1-02

### PR DESCRIPTION
**Pull request by rkmanda**
Add linter rule for RPC-Get-V1-02 - Request body for Get must be empty.

### CODE REVIEW

File: docs/get-must-not-have-request-body.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# GetMustNotHaveRequestBody
+
+## Category
+
+SDK Error
```
Comment (jianyexi) [inline docs/get-must-not-have-request-body.md:5]: ARM Error

File: docs/get-must-not-have-request-body.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Applies to
+
+ARM and Data plane OpenAPI(swagger) specs
```
Comment (jianyexi) [inline docs/get-must-not-have-request-body.md:9]: applied to dataplane specs ?

File: packages/rulesets/src/spectral/az-arm.ts [hunk 3]
```diff
@@ -147,6 +147,18 @@ const ruleset: any = {
...
     ///
 
+    // RPC Code: RPC-Get-V1-02
+    GetMustNotHaveRequestBody: {
+      description: "The Get operation must not have a request body.",
+      severity: "error",
+      message: "{{description}}",
+      resolved: true,
+      formats: [oas2],
+      given: "$.paths.*.get.parameters[?(@.in === 'body')]",
```
Comment (jianyexi) [inline packages/rulesets/src/spectral/az-arm.ts:157]:
```suggestion
      given: "$[paths,'x-ms-paths'].*.get.parameters[?(@.in === 'body')]",
```

Comment (rkmanda) [reply packages/rulesets/src/spectral/az-arm.ts:157]: whats the difference between these 2?

Comment (rkmanda) [reply packages/rulesets/src/spectral/az-arm.ts]: Is it better perf to do it this way?

Comment (jianyexi) [reply packages/rulesets/src/spectral/az-arm.ts]: just to cover the paths under 'x-ms-paths'

**Review by jianyexi — APPROVED** [review]

**Pull request merged**